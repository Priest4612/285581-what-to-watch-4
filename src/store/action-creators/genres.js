import ActionType from '../action-type';

const genresLoaded = (genres) => ({
  type: ActionType.Genres.FETCH_GENRES_SUCCESS,
  payload: genres,
});

const setActiveGenre = (name) => ({
  type: ActionType.Genres.CHANGE_ACTIVE_GENRE,
  payload: name,
});

export default {
  genresLoaded,
  setActiveGenre,
};
