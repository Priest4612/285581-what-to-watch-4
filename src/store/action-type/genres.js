export default {
  FETCH_GENRES_FAILURE: `FETCH_GENRES_FAILURE`,
  FETCH_GENRES_SUCCESS: `FETCH_GENRES_SUCCESS`,
  FETCH_GENRES_REQUEST: `FETCH_MOVIES_REQUEST`,
  CHANGE_ACTIVE_GENRE: `CHANGE_ACTIVE_GENRE`,
};
