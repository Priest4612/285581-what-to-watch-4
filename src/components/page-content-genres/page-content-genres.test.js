import React from 'react';
import renderer from 'react-test-renderer';

import genres from '../../mocks/test-genres-mock.json';

import PageContentGenres from './page-content-genres.jsx';


it(`Render PageContentGenres`, () => {
  const tree = renderer
    .create(<PageContentGenres
      onClickGenreItem={() => {}}
      activeGenre={genres[0].name}
      genres={genres}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
