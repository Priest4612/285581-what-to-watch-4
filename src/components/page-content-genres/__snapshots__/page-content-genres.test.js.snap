// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PageContentGenres 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item
            catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Все жанры
    </a>
  </li>
  <li
    className="catalog__genres-item
            false"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Криминал
    </a>
  </li>
  <li
    className="catalog__genres-item
            false"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Исторический
    </a>
  </li>
  <li
    className="catalog__genres-item
            false"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Триллер
    </a>
  </li>
  <li
    className="catalog__genres-item
            false"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Короткометражный
    </a>
  </li>
  <li
    className="catalog__genres-item
            false"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Мистика
    </a>
  </li>
  <li
    className="catalog__genres-item
            false"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Вестерн
    </a>
  </li>
  <li
    className="catalog__genres-item
            false"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Ужасы
    </a>
  </li>
  <li
    className="catalog__genres-item
            false"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Спорт
    </a>
  </li>
</ul>
`;
